src/http/httpd.cc

```diff
@@ -423,7 +427,8 @@ future<> http_server::do_accepts(int which) {
 
 future<> http_server::do_accept_one(int which) {
     return _listeners[which].accept().then([this] (accept_result ar) mutable {
-        auto conn = std::make_unique<connection>(*this, std::move(ar.connection), std::move(ar.remote_address));
+        auto conn = std::make_unique<connection>(*this,
+            std::move(ar.connection), std::move(ar.remote_address));
```

Comment:
This calls deprecated `connection(server, fd, address)` constructor which ignores the `remote_address` argument and goes calling methods on fd.

@tchaikov , something is wrong with 3841bbdd19e8fb879c266d02a6f1ffbf3cb6f8ee -- this place in httpd.cc still uses the deprecated constructor (even before patching), but compiler silently allows that

Reply:
Except for the white space I erroneously added, this was how `connection` was constructed before my PR

Reply:
I can change it to call the new constructor

Reply:
> I can change it to call the new constructor

The plan was to 
- get remote address from accept_result
- in case connection is constructed without accept_result at hand -- call the fd.remote_address()

Reply:
> This calls deprecated `connection(server, fd, address)` constructor which ignores the `remote_address` argument and goes calling methods on fd.
> 
> @tchaikov , something is wrong with [3841bbd](https://github.com/scylladb/seastar/commit/3841bbdd19e8fb879c266d02a6f1ffbf3cb6f8ee) -- this place in httpd.cc still uses the deprecated constructor (even before patching), but compiler silently allows that

see #1817 . i wanted to add the compile option to let compiler complains when compiling the caller sites of deprecated symbols. but alas, Avi does not agree with me on this.